stage0/gc.go

```diff
@@ -73,9 +74,15 @@ type mounts []*mount
 // parent. This works by pushing all child mounts up to the front. Parent
 // mounts are halted once they encounter a child. The actual sequential order
 // of the mounts doesn't matter; only that children are in front of parents.
-func (m mounts) Less(i, j int) bool { return m[i].id != m[j].parentId }
-func (m mounts) Len() int           { return len(m) }
-func (m mounts) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
+// func (m mounts) Less(i, j int) bool { return m[i].id != m[j].parentId && m[i].parentId > m[j].parentId }
+func (m mounts) Less(i, j int) bool {
+	isNotParent := m[i].id != m[j].parentId
+	hasHigherParent := m[i].parentId > m[j].parentId
```

Comment:
Agreed. Ideally we have a tree structure which allows climbing up and down to detect relations.